### PR TITLE
Add HTML processing and WordPress publishing with example

### PR DESCRIPTION
**Pull request by michplunkett**
## What does this PR do?
It adds a command and how-to text in the `README` for processing local `index.html` file, which will house our visuals, into a single minified file.

## Testing
- [x] Put end result of `npm run process-visual --year=2023 --story=example-visual` into the repository
- [x] Manually ran the command, please see below for proof.

```
michaelp@MacBook-Air-3 chicagomaroon.github.io % ls ./2023/example-visual                                 
index.html	main.css	main.js		meta_data.json
michaelp@MacBook-Air-3 chicagomaroon.github.io % npm run process-visual --year=2023 --story=example-visual

> chicagomaroon.github.io@1.0.0 process-visual
> npx htmlprocessor ./$npm_config_year/$npm_config_story/index.html -o ./$npm_config_year/$npm_config_story/index.proc.html && npx minify ./$npm_config_year/$npm_config_story/index.proc.html > ./$npm_config_year/$npm_config_story/$npm_config_story.min.html && rm ./$npm_config_year/$npm_config_story/index.proc.html

File "./2023/example-visual/index.proc.html" created.
michaelp@MacBook-Air-3 chicagomaroon.github.io % ls ./2023/example-visual                                 
example-visual.min.html	index.html		main.css		main.js			meta_data.json
michaelp@MacBook-Air-3 chicagomaroon.github.io % 
```

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -3,5 +3,6 @@
 jquery*
 *bootstrap*
 README.md
+*index.min.html
```
Comment (michplunkett) [inline .prettierignore:6]: Prettier does not need to run on minified `index.html` files.

File: 2023/example-visual/index.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<html>
```
Comment (michplunkett) [inline 2023/example-visual/index.html:1]: How-to example file for new visuals.

File: 2023/example-visual/index.min.html [hunk 3]
```diff
@@ -0,0 +1 @@
...Alphas:.6,lineAlpha:.4,title:"Assault",valueField:"assault"},{balloonText:"Battery <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Battery",valueField:"battery"},{balloonText:"Burglary <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Burglary",valueField:"burglary"},{balloonText:"Theft <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Theft",valueField:"theft"},{balloonText:"Motor vehicle theft <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Motor vehicle theft",valueField:"motor vehicle theft"},{balloonText:"Arson <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Arson",valueField:"arson"}],plotAreaBorderAlpha:0,marginTop:10,marginLeft:0,marginBottom:0,chartScrollbar:{},chartCursor:{cursorAlpha:0},categoryField:"year",categoryAxis:{startOnAxis:!0,axisColor:"#DADADA",gridAlpha:.07,title:"Year"},export:{enabled:!0}})</script><style>#chart-div{width:100%;height:500px;font-size:11px}</style><div id=chart-div></div>
```
Comment (michplunkett) [inline 2023/example-visual/index.min.html:1]: Output form processing.

File: package.json [hunk 4]
```diff
@@ -4,7 +4,9 @@
     "description": "Feature stories by The Chicago Maroon grouped by year into corresponding folders.",
     "main": "index.js",
     "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1"
+        "format": "npx prettier --write .",
+        "lint": "npx eslint .",
+        "process-visual": "npx htmlprocessor .$npm_config_dir/index.html -o .$npm_config_dir/index.proc.html && npx minify .$npm_config_dir/index.proc.html > .$npm_config_dir/index.min.html && rm .$npm_config_dir/index.proc.html"
```
Comment (michplunkett) [inline package.json:9]: `npm` commands for working with this respository.

File: README.md [hunk 5]
```diff
@@ -3,14 +3,31 @@
...
 
 ## Tools needed
-
 1. [npm](https://www.npmjs.com/) v7.20.3
 
+## Libraries to read when creating visuals
+1. `node-htmlprocessor`: https://github.com/dciccale/node-htmlprocessor
+2. `amCharts`: https://www.amcharts.com/
+   - We are currently using the non-profit license that requires that the `amCharts` logo be displayed on the chart.
+   - If any questions come from the `amCharts` or `UChicago` legal department regarding this license, please reference this `amCharts` support request number: `75825`
```
Comment (michplunkett) [inline README.md:19]: License info, @ywlu026.

File: README.md [hunk 6]
```diff
@@ -3,14 +3,31 @@
...
+3. Create your visual in a file called `index.html`.
+   - Look to the documentation [here](https://github.com/dciccale/grunt-processhtml#readme) for how to write your processed HTML.
+   - Note that any external files, things not stored within this repository, should not be included in the `<build>` tags.
+4. Run `npm run lint` and `npm run format` and make the changes that the output of those commands recommend, if they recommend anything.
+5. Go to the base directory of the `chicagomaroon.github.io` repository and run the command `npm run process-html --dir=[/story/directory]`
+   - Example: `npm run process-visual --dir=/2023/example-visual`
+6. After you have run the command you should see a file named `index.min.html` in your specified directory which can be passed into the website storage bin and used within an [iframe](https://www.w3schools.com/tags/tag_iframe.ASP) to display your story.
+   - This process allows you to editorial an HTML file contains all the data necessary for the display of an interactive visual.
```
Comment (michplunkett) [inline README.md:27]: Read over this if you could, @ywlu026, and just to confirm that it's intelligible beyond just my own 🧠.

File: README.md [hunk 7]
```diff
@@ -3,14 +3,31 @@
...
 
-## Notable commands
+## Commands
+- `npm run format`: This will format the JavaScript inside the repository
+- `npm run lint`: Runs the linter in the repository and will let you know if any JavaScript faux pas were made in your code
+- `npm run process-visual --dir=/year/story-name`: Runs `node-htmlprocessor` on the `index.html` file in the `/year/story-name` directory and outputs the processed version as `index.min.html`
```
Comment (michplunkett) [inline README.md:32]: This is meant for a technical audience, but I'd appreciate you reading over this as well.

File: .prettierignore [hunk 8]
```diff
@@ -3,5 +3,6 @@
 jquery*
 *bootstrap*
 README.md
+*.min.html
```
Comment (michplunkett) [inline .prettierignore:6]: Prettier does not need to run on minified `*.min.html` files.

File: 2023/example-visual/example-visual.min.html [hunk 9]
```diff
@@ -0,0 +1 @@
...Alphas:.6,lineAlpha:.4,title:"Assault",valueField:"assault"},{balloonText:"Battery <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Battery",valueField:"battery"},{balloonText:"Burglary <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Burglary",valueField:"burglary"},{balloonText:"Theft <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Theft",valueField:"theft"},{balloonText:"Motor vehicle theft <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Motor vehicle theft",valueField:"motor vehicle theft"},{balloonText:"Arson <span style='font-size:14px; color:#000000;'><b>[[value]]</b></span>",fillAlphas:.6,lineAlpha:.4,title:"Arson",valueField:"arson"}],plotAreaBorderAlpha:0,marginTop:10,marginLeft:0,marginBottom:0,chartScrollbar:{},chartCursor:{cursorAlpha:0},categoryField:"year",categoryAxis:{startOnAxis:!0,axisColor:"#DADADA",gridAlpha:.07,title:"Year"},export:{enabled:!0}})</script><style>#chart-div{width:100%;height:500px;font-size:11px}</style><div id=chart-div></div>
```
Comment (michplunkett) [inline 2023/example-visual/example-visual.min.html:1]: Output of the example minified visual.

File: README.md [hunk 10]
```diff
@@ -3,14 +3,31 @@
...
+5. Go to the base directory of the `chicagomaroon.github.io` repository and run the command `npm run process-html --year=2023 --name=[kebab-case-story-name]`
+   - Example: `npm run process-visual --year=2023 --name=example-visual`
+6. After you have run the command you should see a file named `index.min.html` in your specified directory which can be passed into the website storage bin and used within an [iframe](https://www.w3schools.com/tags/tag_iframe.ASP) to display your story.
+   - This process allows you to editorial an HTML file contains all the data necessary for the display of an interactive visual.
 
-## Notable commands
+## Commands
+- `npm run format`: This will format the JavaScript inside the repository
+- `npm run lint`: Runs the linter in the repository and will let you know if any JavaScript faux pas were made in your code
+- `npm run process-visual --year=[year] --name=[kebab-case-story-name]`: Runs `node-htmlprocessor` on the `index.html` file in the `/year/story-name` directory and outputs the processed version as `story-name.min.html`
```
Comment (michplunkett) [inline README.md]: A glance over of these `npm` commands would be helpful as well, @ywlu026.

File: README.md [hunk 11]
```diff
@@ -3,14 +3,31 @@
...
+3. Create your visual in a file called `index.html`.
+   - Look to the documentation [here](https://github.com/dciccale/grunt-processhtml#readme) for how to write your processed HTML.
+   - Note that any external files, things not stored within this repository, should not be included in the `<build>` tags.
+4. Run `npm run lint` and `npm run format` and make the changes that the output of those commands recommend, if they recommend anything.
+5. Go to the base directory of the `chicagomaroon.github.io` repository and run the command `npm run process-html --year=2023 --name=[kebab-case-story-name]`
+   - Example: `npm run process-visual --year=2023 --name=example-visual`
+6. After you have run the command you should see a file named `index.min.html` in your specified directory which can be passed into the website storage bin and used within an [iframe](https://www.w3schools.com/tags/tag_iframe.ASP) to display your story.
+   - This process allows you to editorial an HTML file contains all the data necessary for the display of an interactive visual.
```
Comment (michplunkett) [inline README.md:27]: Read over this if you could, @ywlu026, to confirm that it's intelligible beyond just my own 🧠.

File: README.md [hunk 12]
```diff
@@ -3,14 +3,31 @@
...
```
Comment (michplunkett) [inline README.md]: Read over this if you could, @ywlu026, to confirm that it's intelligible beyond just my own 🧠.

File: 2023/example-visual/meta_data.json [hunk 13]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (michplunkett) [inline 2023/example-visual/meta_data.json:1]: Metadata template.

File: export_visual_to_wordpress.sh [hunk 14]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
```
Comment (michplunkett) [inline export_visual_to_wordpress.sh]: This feature is very much a work in progress, but I'm gonna keep it here for future's sake. There's no private information in here.

Comment (michplunkett) [reply export_visual_to_wordpress.sh]: With some time apart from this, I think I'll just do it using node as that's the most relevant tech to this stack and it's so much easier to debug.

**Review by ywlu026 — APPROVED** [review]

**Comment by michplunkett** [comment]
I'm gonna add the WordPress upload feature to this branch since I'm already here. I'll merge it when it's done and show some proof it works in the notes.

Edit: Done!

File: README.md [hunk 15]
```diff
@@ -2,15 +2,47 @@
...
+     "title": "Example Visual",
+     "author": "FirstName LastName",
+     "caption": "Here is some information about the visual.",
+     "description": "Describe your visual"
+   }
+   ```
+5. Run `npm run lint` and `npm run format` and make the changes that the output of those commands recommend, if they recommend anything.
+6. Go to the base directory of the `chicagomaroon.github.io` repository and run the command `npm run process-html --year=2023 --story=[kebab-case-story-name]`
+   - Example: `npm run process-visual --year=2023 --story=example-visual`
+7. You should now have a file named `[story].min.html` in your story's folder. Run the command `npm run wordpress-upload --year=2023 --story=[kebab-case-story-name]` and your `[story].min.html` file will be uploaded to the `media` folder on the Chicago Maroon WordPress site.
```
Comment (michplunkett) [inline README.md]: Need to fix an error here.

File: upload_to_wordpress.js [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+    endpoint: 'https://chicagomaroon.com/wp-json',
+    username: process.env.CHICAGO_MAROON_USER_NAME,
+    password: process.env.CHICAGO_MAROON_PASSWORD
+});
+
+wp.media()
+  .file(`${argv.a}/${argv.b}`)
+  .create()
+  .then(r => {
+    // Your media is now uploaded: let's associate it with a post
```
Comment (michplunkett) [inline upload_to_wordpress.js]: Need to add user information once we are able to add .html files.

**Pull request merged**